File: content/showcase/13-struct-and-namespace/struct-and-namespace.lacking.zig
// lacking: `Point.dims` cannot be read through a value
// a struct is also a namespace: a decl inside it is
// read through the type, never through a value
const std = @import("std");

const Point = struct {
    x: i32,
    y: i32,
    const dims: i32 = 2;
};

pub fn main() void {
    const p = Point{ .x = 1, .y = 2 };
    std.debug.print("{d} {d}\n", .{p.x, Point.dims});
}

File: content/showcase/13-struct-and-namespace/struct-and-namespace.lacking.rs
// lacking: `Point::DIMS` cannot be read through a value
// a struct holds the fields; an associated const is
// read through the type only, never through a value
struct Point { x: i32, y: i32 }
impl Point { const DIMS: i32 = 2; }

fn main() {
    let p = Point { x: 1, y: 2 };
    println!("{} {}", p.x + p.y, Point::DIMS); // 3 2
}
